forc-pkg/src/pkg.rs

```diff
@@ -2529,6 +2553,24 @@ pub fn check(
         let pkg = &plan.graph[node];
         let manifest = &plan.manifest_map()[&pkg.id()];
 
+        // Setup experimental features
```

Comment:
This is quite some copy-paste. Can we have a single function doing it? Actually, the comment already sounds like a perfect function name: `setup_experimental_features` :smile:

Reply:
Also, thinking of encapsulation, shouldn't that function be in the `sway_features::ExperimentalFeatures`? In that case, all the existing functions would actually be private and the tools would just be calling `ExperimentalFeatures::setup(manifest, experimental, no_experimental)` instead of `default()` and not bother about the internal logic, ordering etc. as they must now.

Reply:
Done.